pkg/asset/cluster/gcp/permissions.go

```diff
@@ -0,0 +1,50 @@
+package gcp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	api "google.golang.org/api/cloudresourcemanager/v1"
+)
+
+var (
+	searchPermissions = []string{
+		"compute.firewalls.create",
+	}
+)
+
+// CloudPermissions is a struct that will contain the permission fields mapped to a boolean
+// value where the permission either exists or does not for the Service account in regard
+// to a specific project
+type CloudPermissions struct {
+	CreateFirewallRules bool `json:"compute.firewalls.create"`
+}
+
+// ValidatePermissions will pass a set of permissions to the cloud resource manager api and
+// determine which permissions were found from the original list. When a permission is not listed
+// in the returned data, it is not a permission of the current user in the supplied project.
+func ValidatePermissions(project string) (*CloudPermissions, error) {
+	ctx := context.Background()
+	service, err := api.NewService(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create resource manager service")
+	}
+
+	projectsService := api.NewProjectsService(service)
+	rb := &api.TestIamPermissionsRequest{Permissions: searchPermissions}
+	response, err := projectsService.TestIamPermissions(project, rb).Context(ctx).Do()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve permissions for project %s", project)
```

Comment:
I think this code (creation of the service and the API call) should live in https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/gcp/client.go and follow the patterns there. It's unclear to me how the existing code is authenticating, but moving to client.go the pattern with using the session credentials should work fine.